In [1]:
# Importing libraries 
import numpy as np 
import pandas as pd

In [3]:
conda install -c conda-forge statsmodels

Solving environment: ...working... done

## Package Plan ##

  environment location: C:\ProgramData\Anaconda3

  added / updated specs:
    - statsmodels


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.14.0               |   py39hcbf5309_0         1.0 MB  conda-forge
    python_abi-3.9             |           2_cp39           4 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         1.0 MB

The following NEW packages will be INSTALLED:

  python_abi         conda-forge/win-64::python_abi-3.9-2_cp39

Note: you may need to restart the kernel to use updated packages.




==> WARNING: A newer version of conda exists. <==
  current version: 4.12.0
  latest version: 24.9.1

Please update conda by running

    $ conda update -n base -c defaults conda



EnvironmentNotWritableError: The current user does not have write permissions to the target environment.
  environment location: C:\ProgramData\Anaconda3





The following packages will be UPDATED:

  conda              pkgs/main::conda-4.12.0-py39haa95532_0 --> conda-forge::conda-4.14.0-py39hcbf5309_0




python_abi-3.9       | 4 KB      |            |   0% 
python_abi-3.9       | 4 KB      | ########## | 100% 
python_abi-3.9       | 4 KB      | ########## | 100% 

conda-4.14.0         | 1.0 MB    |            |   0% 
conda-4.14.0         | 1.0 MB    | 1          |   2% 
conda-4.14.0         | 1.0 MB    | ########## | 100% 
conda-4.14.0         | 1.0 MB    | ########## | 100% 
Preparing transaction: ...working... done
Verifying transaction: ...working... failed


In [ ]:
#4.8, 5, 6.4, 6.3,4.4, 4.9, 5.8, 6,
#4.4, 5.2, 6.2, 6.4,4.2, 5.3, 6.2, 4.9,
#3.2, 5.6, 4.7, 5.6,3.8, 5.7, 6.3, 4.6,
#3.9, 4.3, 5.5, 4.8,3.7, 5.4, 6.5, 5.6,
#4.4, 4.8, 5.8, 5.8,3.9, 4.8, 5.5, 5.5

In [49]:
# Create a dataframe 
df = pd.DataFrame({'Watering': np.repeat(['daily', 'weekly'], 20),
                   'Sunlight': np.tile(np.repeat(['None','low', 'med', 'high'], 5), 2),
                   'height': [4.8, 5, 6.4, 6.3,4.4, 4.9, 5.8, 6,
4.4, 5.2, 6.2, 6.4,4.2, 5.3, 6.2, 4.9,

3.2, 5.6, 4.7, 5.6,3.8, 5.7, 6.3, 4.6,

3.9, 4.3, 5.5, 4.8,3.7, 5.4, 6.5, 5.6,

4.4, 4.8, 5.8, 5.8,3.9, 4.8, 5.5, 5.5 ]})

In [50]:
df[:20]

Watering Sunlight  height
0     daily     None     4.8
1     daily     None     5.0
2     daily     None     6.4
3     daily     None     6.3
4     daily     None     4.4
5     daily      low     4.9
6     daily      low     5.8
7     daily      low     6.0
8     daily      low     4.4
9     daily      low     5.2
10    daily      med     6.2
11    daily      med     6.4
12    daily      med     4.2
13    daily      med     5.3
14    daily      med     6.2
15    daily     high     4.9
16    daily     high     3.2
17    daily     high     5.6
18    daily     high     4.7
19    daily     high     5.6

In [53]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
#perform two-way ANOVA
model = ols('height ~ C(Sunlight) + C(Watering) + C(Watering):C(Sunlight)', data=df).fit()
result=sm.stats.anova_lm(model, typ=2)
print(result)

                           sum_sq    df         F    PR(>F)
C(Sunlight)               2.15475   3.0  0.934917  0.435195
C(Watering)               0.60025   1.0  0.781321  0.383331
C(Watering):C(Sunlight)   1.12075   3.0  0.486278  0.694211
Residual                 24.58400  32.0       NaN       NaN


The watering p-value is equal to 0.435195
The Watering p-value is equal to 0.383331
The Fertilizer * Watering: p-value is equal to 0.1668238
The p-values for water and sun turn out to be less than 0.05 which implies that the means of both the factors possess a statistically significant effect on plant height. The p-value for the interaction effect (0.1668238) is greater than 0.05 which depicts that there is no significant interaction effect between fertilizer frequency and watering frequency.